src/components/app/DocumentList/index.js

```diff
@@ -0,0 +1,855 @@
+      isIncluded,
+      fetchDocument,
+      indicatorState,
+      selectTableItems,
+      updateRawModal,
+      viewId,
+    } = this.props;
+
+    indicatorState('pending');
+
+    if (updateUri) {
+      id && updateUri('viewId', id);
+      page && updateUri('page', page);
+      sortingQuery && updateUri('sort', sortingQuery);
+    }
+
+    return fetchDocument(windowType, id, page, this.pageLength, sortingQuery)
+      .then((response) => {
+        const result = response.result;
+        const resultById = {};
+        const selection = getSelectionDirect(selections, windowType, viewId);
+        const forceSelection =
+          (type === 'includedView' || isIncluded) &&
+          response &&
+          result.length > 0 &&
+          (selection.length === 0 ||
+            !doesSelectionExist({
+              data: result,
+              selected: selection,
+            }));
+
+        result.map((row) => {
+          const parsed = parseToDisplay(row.fieldsByName);
+          resultById[`${row.id}`] = parsed;
+          row.fieldsByName = parsed;
+        });
+
+        const pageColumnInfosByFieldName = response.columnsByFieldName;
+
+        mergeColumnInfosIntoViewRows(
+          pageColumnInfosByFieldName,
+          response.result
+        );
+
+        if (this.mounted) {
+          const newState = {
+            pageColumnInfosByFieldName: pageColumnInfosByFieldName,
+            triggerSpinner: false,
+          };
+
+          if (response.filters) {
+            newState.filtersActive = filtersToMap(response.filters);
+          }
+
+          if (
+            locationAreaSearch ||
+            (newState.filtersActive &&
+              newState.filtersActive.has(`location-area-search`))
+          ) {
+            this.getLocationData(resultById);
+          }
+
+          this.setState({ ...newState }, () => {
+            if (forceSelection && response && result && result.length > 0) {
+              const selection = [result[0].id];
+
+              selectTableItems({
+                windowType,
+                viewId,
+                ids: selection,
+              });
+            }
+          });
+
+          // process modal specific
+          const { parentViewId, parentWindowId, headerProperties } = response;
+
+          updateRawModal(windowType, {
+            parentViewId,
+            parentWindowId,
+            headerProperties,
+          });
+        }
+
+        indicatorState('saved');
+      })
+      .catch(() => {
+        this.setState({ triggerSpinner: false });
+      });
+  };
+
+  getLocationData = (resultById) => {
+    const {
+      windowType,
+      viewId,
+      reduxData: { mapConfig },
+    } = this.props;
+
+    locationSearchRequest({ windowId: windowType, viewId }).then(({ data }) => {
+      const locationData = data.locations.map((location) => {
+        const name = get(
+          resultById,
+          [location.rowId, 'C_BPartner_ID', 'value', 'caption'],
```

Comment:
is this ok for this to be hardcoded ? I saw it a couple of times by now. I believe is time to have it in a constant somewhere. Thanks

Reply:
No idea tbh. We have those in many places but can this be changed right now ? I doubt it.

Reply:
![Screenshot 2020-03-03 at 19 49 02](https://user-images.githubusercontent.com/1708561/75804062-1eafe200-5d88-11ea-83ba-c89fe6ad532e.png)
There are two places where it is used afais. Adding one constant and import it in the two files. 5 min job. Saves time in the future. ;)